compiler/src/iree/compiler/Dialect/HAL/Target/CUDA/CUDATarget.cpp

```diff
@@ -383,6 +463,19 @@ class CUDATargetBackend final : public TargetBackend {
                      variantOp.getName(), ".ptx", ptxImage);
     }
 
+    // Attempt compiling the generated PTX with ptxas. If it fails for some
+    // reason fallback to runtime compilation.
+    if (!clUsePTXASfrom.empty()) {
+      FailureOr<std::string> maybePtxImage = compileWithPTXAS(
+          clUsePTXASfrom, clTargetChip, clUsePTXASparams, ptxImage);
+      if (succeeded(maybePtxImage)) {
+        ptxImage = maybePtxImage.value();
```

Comment:
how does the runtime know whether it is getting ptx or cubin? It needs to call a different API right?

Reply:
ah cool, I didn't know

Reply:
> By the way we can also embed the ptx in vmfb as well, so if the binary in `cubin` is not suitable for the target GPU, we can jit compile.

yeah let's keep it simple for now